thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp

```diff
@@ -1791,6 +1827,23 @@ class CKeyedJoinSlave : public CSlaveActivity, implements IJoinProcessor, implem
                 diskSeeks++;
             }
         }
+        virtual void getSubFileStats(std::vector<OwnedPtr<CRuntimeStatisticCollection>> & subFileStats) override
+        {
+            ISuperFileDescriptor * superFDesc = allParts->item(0).queryOwner().querySuperFileDescriptor();
+            for (size_t i=0; i<parts.size(); i++)
+            {
+                unsigned partNo = parts[i] & partMask;
+                unsigned subfile, lnum;
+                PartIO partIO = partIOs[i];
+                if (partIO.iFileIO)
```

Comment:
Shall I be using queryFetchPartIO ?  I.e. Should it be opening the stream if it's not already open? 

Can you also clarify why the atomic is needed?  Can the partIO.iFileIO be closed by another thread?

Reply:
>Shall I be using queryFetchPartIO ? I

no, because that would created it, and there's no need for this thread to, there will be no stats. if not created.

>Can you also clarify why the atomic is needed? Can the partIO.iFileIO be closed by another thread?

it can't be closed, but it's not strictly thread safe for 1 thread to write a pointer, whilst another concurrently reads it.
In reality on Intel architectures it will be fine, but not necessarily on all, on some it could write 1st 32bit of pointer, and the reading thread uses that before fully initialized.